### ISSUE TITLE
About Penalized logistic regression did not converge in REGENIE v3.1.1

### ISSUE THREAD

**Issue by Renown-TAL** [rgcgithub/regenie#310]
Hi, 

I get the warnning with the Options as following:
_________________________________________________________________________________
 Options in effect:
  --step 1 \
  --force-step1 \
  --niter 1000 \
  --bed ukb23149_c20_b0_v1 \
  --covarFile covar_2.txt \
  --phenoFile phenotypes.txt \
  --bsize 1000 \
  --bt \
  --lowmem \
  --lowmem-prefix ukb23150_c20_b0_v1 \
  --out ukb23150_c20_b0_v1 \
  --extract ukb23149_c20_b0_v1.snplist
Fitting null model
 * bim              : [ukb23149_c20_b0_v1.bim] n_snps = 677089
   -keeping variants specified by --extract
   -number of variants remaining in the analysis = 30663
 * fam              : [ukb23149_c20_b0_v1.fam] n_samples = 454756
 * bed              : [ukb23149_c20_b0_v1.bed]
 * phenotypes       : [phenotypes.txt] n_pheno = 1
   -dropping observations with missing values at any of the phenotypes
   -number of phenotyped individuals with no missing data = 438187
 * covariates       : [covar_2.txt] n_cov = 4
   -number of individuals with covariate data = 438187
 * number of individuals used in analysis = 438187
 * case-control counts for each trait:
   - 'PD': 3170 cases and 435017 controls
   -fitting null logistic regression on binary phenotypes...done (267ms) 
   -residualizing and scaling phenotypes...done (4ms) 
 * # threads        : [15]
 * block size       : [1000]
 * # blocks         : [31] for 30663 variants
 * # CV folds       : [5]
 * ridge data_l0    : [5 : 0.01 0.25 0.5 0.75 0.99 ]
 * ridge data_l1    : [5 : 0.01 0.25 0.5 0.75 0.99 ]
 * approximate memory usage : 4GB
 * writing level 0 predictions to disk
   -temporary files will have prefix [ukb23150_c20_b0_v1_l0_Y]
   -approximate disk space needed : 538MB
_________________________________________________________________________
the warning:
_________________________________________________________________________
-on phenotype 1 (PD)...WARNING: Penalized logistic regression did not converge! (Increase --niter)
done (1338922ms) 
Output
------
phenotype 1 (PD) : Level 1 model did not converge. LOCO predictions calculations are skipped.
_________________________________________________________________________
and my phenotypes case ratio is 3537:480714, and covar is sex,age,age^2,sex-by-age.

I get the .loco file with other chr, but fail in chr20 and chr4. I want to know should i Increase the niter(like10000 and more), or change some options?

Thanks in advance!

**Comment by joellembatchou** [comment]
Hi,

I don't understand what you mean by getting loco predictions only for specific chromosomes. The logistic regression is only fit once for all chromosomes so if it does not converge then you should not get any predictions. Are you fitting step 1 separately for each chromosome?

Cheers,
Joelle

**Comment by Renown-TAL** [comment]
> Hi,
> 
> I don't understand what you mean by getting loco predictions only for specific chromosomes. The logistic regression is only fit once for all chromosomes so if it does not converge then you should not get any predictions. Are you fitting step 1 separately for each chromosome?
> 
> Cheers, Joelle

Yes, we fitting step 1 separately for each chromosome. Because we do this work in ukbiobank dnanexus, runing all chromosomes-merge file will spend more time and money. It is necessary to test all chromosomes-merge file?

**Comment by joellembatchou** [comment]
Hi,

The goal of step 1 is to capture **genome-wide** polygenic effects with a smaller set of variants (e.g. directly genotyped SNPs) so you should run it with all the autosomal chromosomes (ie create a merged file with all chrs and pass that to Regenie).

Cheers,
Joelle
